### PR TITLE
PLT-119 First draft of outgoing webhook documentation

### PR DESCRIPTION
**Pull request by jwilander**
Looking for feedback.

Also made some minor changes to the incoming webhook doc

### CODE REVIEW

File: doc/integrations/webhooks/Outgoing-Webhooks.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+A couple key points:
+
+- **Mattermost outgoing webhooks are Slack-compatible.** If you've used Slack's outgoing webhooks to create integrations, you can copy and paste that code to create Mattermost integrations. Mattermost automatically translates Slack's proprietary JSON payload format into markdown to render in Mattermost messages
+- **Mattermost outgoing webhooks support full markdown.** When an integration responds with a message to post, it will have access to a rich range of formatting unavailable in Slack that is made possible through [markdown support](../../usage/Markdown.md) in Mattermost. This includes headings, formatted fonts, tables, inline images and other options supported by [Mattermost Markdown]
+
+_Example:_
+
+Suppose you had an external application that recieved a post event asking for the status of a daily build. The application could respond with a table of total tests run and total tests failed by component category, with links to failed tests by category. An example response the application could respond with would be:
```
Comment (it33) [inline doc/integrations/webhooks/Outgoing-Webhooks.md]: Perhaps show an example of what a user might type to trigger the outgoing webhook?

File: doc/integrations/webhooks/Outgoing-Webhooks.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+### Slack Compatibility
+
+As mentioned above, Mattermost makes it easy to take integrations written for Slack's proprietary JSON payload format and repurpose them to become Mattermost integrations. The following automatic translations are supported:
+
+1. The HTTP POST request body is formatted the same as Slack's, which means your Slack integration's receiving function should not need to change at all to be compatible with Mattermost
+2.  JSON responses designed for Slack using `<>` to note the need to hyperlink a URL, such as ```{"text": "<http://www.mattermost.com/>"}```, are translated to the equivalent markdown in Mattermost and rendered the same as you would see in Slack
+3. Similiarly, responses designed for Slack using `|` within a `<>` to define linked text, such as ```{"text": "Click <http://www.mattermost.com/|here> for a link."}```, are also translated to the equivalent markdown in Mattermost and rendered the same as you would see in Slack
+
+To see samples and community contributions, please visit <http://mattermost.org/webhooks>.
```
Comment (it33) [inline doc/integrations/webhooks/Outgoing-Webhooks.md]: Perhaps add a "Limitations" section for anything Slack supports that Mattermost doesn't yet support? E.g. https://github.com/mattermost/platform/blob/master/doc/integrations/webhooks/Incoming-Webhooks.md#limitations

**Comment by it33** [comment]
Just two small comments, otherwise looks good,

File: doc/integrations/webhooks/Outgoing-Webhooks.md (hunk 2 above)
Comment (jwilander) [inline doc/integrations/webhooks/Outgoing-Webhooks.md]: Added

**Comment by crspeller** [comment]
+1

**Pull request merged**